### ISSUE TITLE
AutocompleteInput using a function as optionText shows [object Object] after selecting an item

### ISSUE THREAD

**Issue by christiaanwesterbeek** [marmelab/react-admin#4281]
**What you were expecting:**
When using an optionRenderer function as the optionText for AutocompleteInput, after selecting an item from the list, I hoped to see something useful in the input.

**What happened instead:**
Instead I see [object Object].
![image](https://user-images.githubusercontent.com/465989/72292934-aca60100-3652-11ea-8d14-9075722b3917.png)

**Steps to reproduce:**
Code like this should do it:
```

const optionRenderer = choice => {
    return (
         <span>{choice.name}-{choice.id}</span>
    )
};
// ...
<ReferenceInput
    source="post_id"
    reference="posts"
    perPage={15}
    sort={{ field: 'title', order: 'ASC' }}
    fullWidth
>
    <AutocompleteInput
        optionText={optionRenderer}
        options={{ fullWidth: true }}
    />
</ReferenceInput>
```

**Related code:**
See https://codesandbox.io/s/pensive-perlman-vwg90 and navigate to comments and select a comment.

**Other information:**
There seems no other prop besides optionText to render a suggestion once it has been selected. So it kinda makes sense that [object Object] is printed, but I consider it a bug if no other means are available to render the selected item. I suppose I'm missing something.

Somewhat related issues:
- https://github.com/marmelab/react-admin/issues/4092
- https://github.com/marmelab/react-admin/issues/4183
- https://github.com/marmelab/react-admin/issues/3503

**Environment**

* React-admin version: 3.1.2
* Last version that did not exhibit the issue (if applicable): not sure
* React version: 16.12
* Browser: any
* Stack trace (in case of a JS error): -

**Comment by djhi** [comment]
Thanks for the report and the sandbox. Reproduced. 

However, note that if you want to render a custom React element, you should do it like this:

```jsx
const OptionRenderer = ({ record )} => (
    <span>
      {choice.title}-{choice.id}
    </span>
);

<AutocompleteInput
    matchSuggestion={(filterValue, suggestion) => true}
    optionText={<OptionRenderer />}
    options={{ fullWidth: true }}
/>
```

But it looks like it is ignored for the currently selected value, which is bug. Besides, the documentation is outdated about this prop.

**Comment by fzaninotto** [comment]
Fixed by #4367
